### ISSUE TITLE
Towards OpenTracing Release

### ISSUE THREAD

**Issue by adriancole** [opentracing/opentracing.github.io#7]

**Comment by adriancole** [comment]
Let's track progress needed before we call out our first version. I'm opening this issue against the documentation site to call out that the api will materialize in various languages, like go, python, java, but docs need to be consistent, and drive them.

My suggestion is that we use github checkboxes to track downstream issues in other projects.. we can also annotate whether or not they are blocking, either visually here, or via a "1.0" or similar milestone in the subordinates.

For example, we have a choice to formalize the api 1.0 before all languages support it, and those languages can implement 1.0 decoupled.

To get started, let's change this description to a mechanism that works (like github checklist)?

@slimsag @yurishkuro @bensigelman wdyt?

**Comment by bensigelman** [comment]
I am permanently grumpy about the state of distributed task management, esp in OSS... but thankfully this effort is still early enough (and the backlog is small enough) that we can get away with just about anything :)

My two cents: I am all in on having a concrete list of things that need to happen before we can send out a 1.0 announcement; I would prefer to wait a bit on defining processes for life much beyond 1.0, though, since I suspect there will be other stakeholders once there's increased awareness and whatever we decide will likely have to change then anyway.

Do both of those points make sense? If so, I am happy to take a stab at populating the list of blocking tasks for a 1.0 release. I'd suggest one Github Issue for each high-level "area" (website, semantic API, individual lang APIs), all pegged to the 1.0 milestone. I'm flexible if that doesn't sound good to people.

**Comment by yurishkuro** [comment]
sgtm.

**Comment by adriancole** [comment]
>
> I am permanently grumpy about the state of distributed task management,
> esp in OSS... but thankfully this effort is still early enough (and the
> backlog is small enough) that we can get away with just about anything :)
>
understood.

> My two cents: I am all in on having a concrete list of things that need to
> happen before we can send out a 1.0 announcement; I would prefer to wait a
> bit on defining processes for life much beyond 1.0, though, since I suspect
> there will be other stakeholders once there's increased awareness and
> whatever we decide will likely have to change then anyway.
>
This isn't so much about process making, more about transparency on what's
actually blocking so others can help asynchronously and those who are eager
for a release can get a sense of what's left (without reading top-to-bottom
or scrolling up via gitter). 1.0 can happen without either of these
aspects, just with remote folks I think this could help.

Usually first release is actually the hardest one as there are so many
tasks. Especially first release, not everything is automated, yet. And what
actually is involved is a work in progress.

Ex here's a list I used in denominator back in the day, which now is fully
automated. I re-updated the description a bunch of times I think.
https://github.com/Netflix/denominator/issues/102

In zipkin, for example, it took us weeks to get a release together, due to
repository, travis etc not working right. I found it easier to spin off
issues and ping people vs try to keep it all in my head. This was
particularly the case since timezones were so different. Ex Often I was
pleasantly surprised to wake up and find things done!

> Do both of those points make sense? If so, I am happy to take a stab at
> populating the list of blocking tasks for a 1.0 release. I'd suggest one
> Github Issue for each high-level "area" (website, semantic API, individual
> lang APIs), all pegged to the 1.0 milestone. I'm flexible if that doesn't
> sound good to people.
>
The burden of managing a first release is something that we'd want to be
less.. so long story short is if you are driving the release and have an
alternate way that serves you better.. I'd say go for it.

I was suggesting things that worked for me, particularly across timezones,
and in no way intended to add process for its own sake.

**Comment by bensigelman** [comment]
I appreciate the clarifications, all makes sense. And I definitely agree that remaining work should be in a structured format somewhere, not in a gitter scroll buffer.

Call me naive, but I think/hope this "1.0" is going to be easier than most since it's truly just APIs without real implementations.

My rough mental model, certainly up for discussion:
1) Get the initial platform APIs and web presence up
2) Soft-launch
3) Make sure we have a few solid tracing systems bound to those APIs where appropriate
4) Make sure that we have a few solid companies committed to using opentracing
5) Add those brands (both the tracing systems and the companies) to the website, perhaps with a pull-quote here or there
6) Non-soft-relaunch :)

To me, 1.0 is just getting us to the first of the steps above. It's basically our way of saying "this is a thing, and the APIs are stable enough to program against." Hopefully by the last step above we will have better automation, etc.

**Comment by bensigelman** [comment]
I added some descriptive color to the "1.0" milestone:

![image](https://cloud.githubusercontent.com/assets/2251553/12180004/f2533096-b52e-11e5-8d16-90539402c040.png)

**Comment by bensigelman** [comment]
@adriancole ok if I close this? I think it's superseded by the "1.0" milestone and #9 ...
